### PR TITLE
[Merged by Bors] - chore: NoZeroSMulDivisors instances for (Add)MonoidAlgebra/Polynomial

### PR DESCRIPTION
**Pull request by alreadydone**
Also generalizes the instance for Finsupp.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Polynomial/Basic.lean [hunk 1]
```diff
@@ -135,6 +135,10 @@ instance smulZeroClass {S : Type*} [SMulZeroClass S R] : SMulZeroClass S R[X] wh
   smul r p := ⟨r • p.toFinsupp⟩
   smul_zero a := congr_arg ofFinsupp (smul_zero a)
 
+instance {S} [Zero S] [SMulZeroClass S R] [NoZeroSMulDivisors S R] : NoZeroSMulDivisors S R[X] where
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Polynomial/Basic.lean:138]:
```suggestion
instance {S : Type*} [Zero S] [SMulZeroClass S R] [NoZeroSMulDivisors S R] : NoZeroSMulDivisors S R[X] where
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Lgtm

File: Mathlib/Algebra/Polynomial/Basic.lean [hunk 2]
```diff
@@ -135,6 +135,10 @@ instance smulZeroClass {S : Type*} [SMulZeroClass S R] : SMulZeroClass S R[X] wh
   smul r p := ⟨r • p.toFinsupp⟩
   smul_zero a := congr_arg ofFinsupp (smul_zero a)
 
+instance {S : Type*} [Zero S] [SMulZeroClass S R] [NoZeroSMulDivisors S R] : NoZeroSMulDivisors S R[X] where
```
Comment (alreadydone) [inline Mathlib/Algebra/Polynomial/Basic.lean:138]:
```suggestion
instance {S : Type*} [Zero S] [SMulZeroClass S R] [NoZeroSMulDivisors S R] :
    NoZeroSMulDivisors S R[X] where
```

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**